var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -54,6 +56,8 @@ class Hdf5(AutotoolsPackage):
             description='Builds a debug version of the library')
     variant('shared', default=True,
             description='Builds a shared version of the library')
+    variant('static', default=True,
```

Comment:
https://spack.readthedocs.io/en/latest/packaging_guide.html?highlight=shared%20static#variant-names
> The semantics of the shared variant are important. When a package is built ~shared, the package guarantees that no shared libraries are built. When a package is built +shared, the package guarantees that shared libraries are built, but it makes no guarantee about whether static libraries are built.

I recall that @alalazo and I have had some discussions about shared/static variants (and multi-value variants) but I don't think we resolved anything. I'd suggest leaving the static variant out.

Reply:
As I understand it, the spack convention and (unofficial?) policy is for a package to build static OR shared but not both, using the `+shared` and `~shared` variant respectively.

Reply:
I've removed the +static variant, and HDF5 (cmake) now builds only shared libs except for when the ~shared variant is specified, and then it builds static libs.  This will be a change from the autotools build, which defaults to build both.